controller/area_blackbox_test.go

```diff
@@ -252,77 +250,48 @@ func (rest *TestAreaREST) createChildArea(name string, parent area.Area, svc *go
-	sp, parentArea := createSpaceAndArea(rest.T(), rest.db)
-	owner, err := rest.db.Identities().Load(context.Background(), sp.OwnerID)
-	require.NoError(rest.T(), err)
-	svc, ctrl := rest.SecuredControllerWithIdentity(owner)
-	rest.createChildArea("TestShowChildrenAreaOKUsingExpiredIfModifedSinceHeader", parentArea, svc, ctrl)
-	// when
-	ifModifiedSince := app.ToHTTPTime(parentArea.UpdatedAt.Add(-1 * time.Hour))
-	res, result := test.ShowChildrenAreaOK(rest.T(), svc.Context, svc, ctrl, parentArea.ID.String(), &ifModifiedSince, nil)
-	//then
-	assert.Equal(rest.T(), 1, len(result.Data))
-	assertResponseHeaders(rest.T(), res)
-}
-
-func (rest *TestAreaREST) TestShowChildrenAreaOKUsingExpiredIfNoneMatchHeader() {
-	// given
-	sp, parentArea := createSpaceAndArea(rest.T(), rest.db)
-	owner, err := rest.db.Identities().Load(context.Background(), sp.OwnerID)
-	require.NoError(rest.T(), err)
-	svc, ctrl := rest.SecuredControllerWithIdentity(owner)
-	rest.createChildArea("TestShowChildrenAreaOKUsingExpiredIfNoneMatchHeader", parentArea, svc, ctrl)
-	// when
-	ifNoneMatch := "foo"
-	res, result := test.ShowChildrenAreaOK(rest.T(), svc.Context, svc, ctrl, parentArea.ID.String(), nil, &ifNoneMatch)
-	//then
-	assert.Equal(rest.T(), 1, len(result.Data))
-	assertResponseHeaders(rest.T(), res)
-}
-
-func (rest *TestAreaREST) TestShowChildrenAreaNotModifiedUsingIfModifedSinceHeader() {
-	// given
-	sp, parentArea := createSpaceAndArea(rest.T(), rest.db)
-	owner, err := rest.db.Identities().Load(context.Background(), sp.OwnerID)
-	require.NoError(rest.T(), err)
-	svc, ctrl := rest.SecuredControllerWithIdentity(owner)
-	childArea := rest.createChildArea("TestShowChildrenAreaNotModifiedUsingIfModifedSinceHeader", parentArea, svc, ctrl)
-	// when
-	ifModifiedSince := app.ToHTTPTime(*childArea.Data.Attributes.UpdatedAt)
-	res := test.ShowChildrenAreaNotModified(rest.T(), svc.Context, svc, ctrl, parentArea.ID.String(), &ifModifiedSince, nil)
-	//then
-	assertResponseHeaders(rest.T(), res)
-}
-
-func (rest *TestAreaREST) TestShowChildrenAreaNotModifiedIfNoneMatchHeader() {
-	// given
-	sp, parentArea := createSpaceAndArea(rest.T(), rest.db)
-	owner, err := rest.db.Identities().Load(context.Background(), sp.OwnerID)
-	require.NoError(rest.T(), err)
-	svc, ctrl := rest.SecuredControllerWithIdentity(owner)
-	childArea := rest.createChildArea("TestShowChildrenAreaNotModifiedIfNoneMatchHeader", parentArea, svc, ctrl)
-	modelChildArea := convertAreaToModel(*childArea)
-	// when
-	ifNoneMatch := app.GenerateEntityTag(modelChildArea)
-	res := test.ShowChildrenAreaNotModified(rest.T(), svc.Context, svc, ctrl, parentArea.ID.String(), nil, &ifNoneMatch)
-	//then
-	assertResponseHeaders(rest.T(), res)
+	childArea := rest.createChildArea("TestShowChildrenArea", *parentArea, svc, ctrl)
+	rest.T().Run("Success", func(t *testing.T) {
+		t.Run("OK", func(t *testing.T) {
+			res, result := test.ShowChildrenAreaOK(rest.T(), svc.Context, svc, ctrl, parentArea.ID.String(), nil, nil)
```

Comment:
Whenever you're in a sub-test, make sure to no longer use `s.T()` but just `t`. Otherwise if a test fails you'll get nasty panics out of nowhere.

Reply:
Yup :+1: